File: Archivos/codigo.py
import os
import subprocess

def abrir_pdf_secuencialmente():
    # Ruta base donde están las carpetas
    ruta_base = r"D:\ESCOM\Semestres\4to Semestre\SO\Ejercicio"

    # Lista de archivos PDF en el orden en que se deben abrir
    archivos_pdf = [
        os.path.join(ruta_base, "Uno", "Ejemplo1.pdf"),
        os.path.join(ruta_base, "Dos", "Ejemplo2.pdf"),
        os.path.join(ruta_base, "Tres", "Ejemplo3.pdf"),
        os.path.join(ruta_base, "Cuatro", "Ejemplo4.pdf"),
    ]
    
    # Abrimos cada archivo secuencialmente
    for archivo in archivos_pdf:
        # Verificamos si el archivo existe antes de intentar abrirlo
        if os.path.exists(archivo):
            print(f"Abriendo {archivo}")
            # Abre el archivo y espera a que se cierre
            with subprocess.Popen([archivo], shell=True) as proceso:
                proceso.wait()  # Espera hasta que el archivo se cierre
            print(f"Cerraste {archivo}")
        else:
            print(f"No se encontró el archivo: {archivo}")

    print("Todos los archivos han sido cerrados. Fin del programa.")

# Ejecutar la función
abrir_pdf_secuencialmente()

File: Archivos/codigo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <windows.h>

// Función para abrir un archivo PDF y esperar a que se cierre
void abrir_pdf(const char *ruta_pdf) {
    // Usamos ShellExecute para abrir el PDF con el programa predeterminado
    HINSTANCE resultado = ShellExecute(NULL, "open", ruta_pdf, NULL, NULL, SW_SHOWNORMAL);

    // Verificamos si el archivo se abrió correctamente
    if (resultado <= (HINSTANCE)32) {
        fprintf(stderr, "Error al intentar abrir el archivo: %s\n", ruta_pdf);
    } else {
        printf("Se abrió el archivo: %s\n", ruta_pdf);
    }

    // Esperar a que el usuario cierre el visor
    printf("Cierra el archivo para continuar...\n");
    getchar();
}

int main() {
    const char *ruta_base = "D:\\ESCOM\\Semestres\\4to Semestre\\SO\\Ejercicio";
    const char *archivos_pdf[] = {
        "\\Uno\\Ejemplo1.pdf",
        "\\Dos\\Ejemplo2.pdf",
        "\\Tres\\Ejemplo3.pdf",
        "\\Cuatro\\Ejemplo4.pdf"
    };

    for (int i = 0; i < 4; i++) {
        char ruta_completa[512];
        snprintf(ruta_completa, sizeof(ruta_completa), "%s%s", ruta_base, archivos_pdf[i]);
        printf("Abriendo archivo: %s\n", ruta_completa);
        abrir_pdf(ruta_completa);
    }

    printf("Todos los archivos han sido cerrados. Fin del programa.\n");
    return 0;
}
